app.go

```diff
@@ -910,7 +910,15 @@ type withTimeoutParams struct {
 
 func withTimeout(ctx context.Context, param *withTimeoutParams) error {
 	c := make(chan error, 1)
-	go func() { c <- param.callback(ctx) }()
+	var wg sync.WaitGroup
+	// Wait for lifecycle goroutine to exit before returning.
+	defer wg.Wait()
```

Comment:
A bit confused; shouldn't the `wg.Wait()` be at line 924 (after the `var err error` decl)?

If the ordering of the code execution is:

context is cancelled while code being executed in `internal/lifecycle/lifecycle.go:79-90`, then `app.go:937` executes before `internal/lifecycle/lifecycle.go:90-96`, we'd be missing records, right?

Shouldn't we wait for the execution of callback entirely before we start executing past `app.go:933`?

Reply:
So yes - in the case you described, it's possible that the record for the hook that was executing when the context timed out is missing. But we don't need the record of that hook because records are specifically for hooks that successfully till the end before a timeout happened. If timeout happened in the middle of executing some hook, then we still report that hook in L953, so I think we should be OK here.